zerver/lib/user_groups.py

```diff
@@ -66,15 +66,26 @@ def has_user_group_access(
     if user_group.is_system_group:
         return False
 
+    can_edit_user_groups_from_realm_level_permission = (
+        user_profile.can_edit_user_groups_from_realm_level_permission()
+    )
+
     group_member_ids = get_user_group_direct_member_ids(user_group)
```

Comment:
We could probably avoid doing this queries when user is admin or moderator, but I think we can leave it as it is for now since we would add a new realm level seting anyway and we would not require this check after that.